src/test/capi_predict_validation_test.cpp

```diff
@@ -0,0 +1,725 @@
+            {"Input_U8_1_3_62_62_NCHW",
+                std::tuple<ovms::shape_t, ovms::Precision>{{1, 3, 62, 62}, ovms::Precision::U8}},
+            {"Input_I64_1_6_128_128_16_NCDHW",
+                std::tuple<ovms::shape_t, ovms::Precision>{{1, 6, 128, 128, 16}, ovms::Precision::I64}},
+            {"Input_U16_1_2_8_4_NCHW",
+                std::tuple<ovms::shape_t, ovms::Precision>{{1, 2, 8, 4}, ovms::Precision::U16}}},
+        {}, putBufferInInputTensorContent, badSize);
+    auto status = instance->mockValidate(&request);
+    EXPECT_EQ(status, ovms::StatusCode::INVALID_CONTENT_SIZE) << status.string();
+}
+
+TEST_F(CAPIPredictValidation, RequestIncorrectContentSizeShapeAuto) {
+    modelConfig.parseShapeParameter("auto");
+    badSize = true;
+    preparePredictRequest(request,
+        {{"Input_FP32_1_224_224_3_NHWC",
+             std::tuple<ovms::shape_t, ovms::Precision>{{1, 224, 224, 3}, ovms::Precision::FP32}},
+            {"Input_U8_1_3_62_62_NCHW",
+                std::tuple<ovms::shape_t, ovms::Precision>{{1, 3, 62, 62}, ovms::Precision::U8}},
+            {"Input_I64_1_6_128_128_16_NCDHW",
+                std::tuple<ovms::shape_t, ovms::Precision>{{1, 6, 128, 128, 16}, ovms::Precision::I64}},
+            {"Input_U16_1_2_8_4_NCHW",
+                std::tuple<ovms::shape_t, ovms::Precision>{{1, 2, 8, 4}, ovms::Precision::U16}}},
+        {}, putBufferInInputTensorContent, badSize);
+    auto status = instance->mockValidate(&request);
+    EXPECT_EQ(status, ovms::StatusCode::INVALID_CONTENT_SIZE) << status.string();
+}
+
+class CAPIPredictValidationInputTensorContent : public ::testing::TestWithParam<ovms::Precision> {
+protected:
+    std::unique_ptr<ov::Core> ieCore;
+    std::unique_ptr<NiceMock<MockModelInstance>> instance;
+    ovms::InferenceRequest request{"model_name", 1};
+    ;
+    ovms::ModelConfig modelConfig{"model_name", "model_path"};
+    ovms::tensor_map_t servableInputs;
+
+    void SetUp() override {
+        ieCore = std::make_unique<ov::Core>();
+        instance = std::make_unique<NiceMock<MockModelInstance>>(*ieCore);
+    }
+};
+
+TEST_P(CAPIPredictValidationInputTensorContent, RequestCorrectContentSizeInputTensorContent) {
+    ovms::Precision testedPrecision = GetParam();
+    const std::string inputName = "someName";
+    servableInputs = ovms::tensor_map_t({
+        {inputName,
+            std::make_shared<ovms::TensorInfo>(inputName, testedPrecision, ovms::shape_t{1, 224, 224, 3}, ovms::Layout{"NHWC"})},
+    });
+    ON_CALL(*instance, getInputsInfo()).WillByDefault(ReturnRef(servableInputs));
+    ON_CALL(*instance, getBatchSize()).WillByDefault(Return(1));
+    ON_CALL(*instance, getModelConfig()).WillByDefault(ReturnRef(modelConfig));
+    preparePredictRequest(request,
+        {{inputName,
+            std::tuple<ovms::shape_t, ovms::Precision>{{1, 224, 224, 3}, testedPrecision}}},
+        {},     // data,
+        true);  // put buffer in InputTensorContent
```

Comment:
this comment is not relevant -> this is not KFS. Is it possible to skip this parameter? This applies to other places in this file as well.

Reply:
Removed.